grid2op/tests/test_alert_trust_score.py

```diff
@@ -0,0 +1,1324 @@
+
+    def attack(self, observation, agent_action, env_action, budget, previous_fails): 
+        if observation is None:
+            return None, None
+        current_step = self.env.nb_time_step
+        if current_step not in self.steps_attack: 
+            return None, None
+        
+        return self.custom_attack, self.duration
+
+
+class TestOpponentMultiLines(GeometricOpponentMultiArea): 
+    """An opponent that can select the line attack, the time and duration of the attack."""
+    
+    def __init__(self, action_space):
+        super().__init__(action_space)
+        self.custom_attack = None
+        self.duration = None
+        self.steps_attack = None
+
+    def init(self, partial_env,  lines_attacked=[ATTACKED_LINE], duration=[10,10], steps_attack=[0,1]):
+        super().init(partial_env, [[l] for l in lines_attacked])
+        attacked_line = lines_attacked[0]
+        self.custom_attack = [ self.action_space({"set_line_status" : [(l, -1)]}) for l in lines_attacked]
+        self.duration = duration
+        self.steps_attack = steps_attack
+        self.env = partial_env
+        
+    def attack(self, observation, agent_action, env_action, budget, previous_fails): 
+        if observation is None:
+            return None, None
+
+        current_step = self.env.nb_time_step
+        if current_step not in self.steps_attack: 
+            return None, None
+        
+        index = self.steps_attack.index(current_step)
+
+        return self.custom_attack[index], self.duration[index]
+
+
+# Test alert blackout / tets alert no blackout
+class TestAlertTrustScoreNoBlackout(unittest.TestCase):
+    """test the basic bahavior of the assistant alert feature when no attack occur """
+
+    def setUp(self) -> None:
+        self.env_nm = os.path.join(
+            PATH_DATA_TEST, "l2rpn_idf_2023_with_alert"
+        )
+
+    def test_assistant_trust_score_no_blackout_no_attack_no_alert(self) -> None : 
+        """ When no blackout and no attack occur, and no alert is raised we expect a reward of 0
+            until the end of the episode where we have a bonus (here artificially 42)
+
+        Raises:
+            Grid2OpException: raise an exception if an attack occur
+        """
+        with make(
+            self.env_nm,
+            test=True,
+            difficulty="1",
+            reward_class=_AlertTrustScore(reward_end_episode_bonus=42)
+        ) as env:
+            env.seed(0)
+            env.reset()
+
+            done = False
+            for i in range(env.max_episode_duration()):
+                obs, score, done, info = env.step(env.action_space())
+                if info["opponent_attack_line"] is None : 
+                    if i == env.max_episode_duration()-1: 
+                        #assert score == 42
+                        assert env._reward_helper.template_reward.total_nb_attacks==0
+                        assert env._reward_helper.template_reward.cumulated_reward==42
```

Comment:
In all of these, can we add asserts for the true positive, false positive etc. ? (4 more assert if i'm not mistaken)